### ISSUE TITLE
Sleep is unreliable in 4.7.0

### ISSUE THREAD

**Issue by TheSpyder** [wavebox/waveboxapp#932]
* Wavebox Version: 4.7.0 (although I think I saw this in one of the recent betas too)
* Operating System & Version: MacOS 10.14.4 beta

**Expected & actual behavior**
Accounts should sleep after the defined timeout, manual sleep should work

**Steps to reproduce**
I don't see this all the time, in particular it works after restarting wavebox, but twice now I've noticed my accounts aren't sleeping (mostly when I look at my memory usage and wavebox has crept up to 3gb). Manual sleep also fails once in this state.

**Is the bug persistent or intermittent?**
intermittent at this stage

**Additional information**
I cleared the developer tools console log and triggered a manual sleep - no message, no sleep.

I'm happy to turn on additional logging flags if that will help track this down.

**Comment by TheSpyder** [comment]
I haven't had time to confirm this but it seems to be after my mac resumes from sleep.

I restarted wavebox right after logging the issue and now it won't sleep again, full developer console log to this point:
```
[Start Metrics]  BS=921 | CE=471 | CS=789 | TO=1728
mailboxes.js:175 [Squirrel] checking for updates
mailboxes.js:12 [SYNCS][WebSocket] Socket opened
mailboxes.js:175 [Squirrel] No updates available
mailboxes.js:12 WebSocket connection to 'wss://cerberus-xxxx.lb.slack-msgs.com/websocket/HwAEz59kqZz6C-ytPshQQKUsI8nzSx9iJh_NJPmjzMOVluUS2B50XAGid4nlXu8_bYPgyp4xTXoZi1ecTHs3tU7gxTjwntEqjwnUwKL9iqc=' failed: Error in connection establishment: net::ERR_INTERNET_DISCONNECTED
open @ mailboxes.js:12
mailboxes.js:12 WebSocket connection to 'wss://cerberus-xxxx.lb.slack-msgs.com/websocket/l5Ent51TQT_0FrIIsmDmEU1x3BLSAN_dSleTJBkb0xgZXzVCpLk7snEX0m950CFeaCZTvPD0QwhhKmo89io9JYUDYZsmsNYFaf0qhsUZyhw=' failed: Error in connection establishment: net::ERR_INTERNET_DISCONNECTED
open @ mailboxes.js:12
mailboxes.js:12 Reconnecting SlackRTM Socket for `0b60f79a-711b-43af-a7b1-c7620e65e7ad` Object
isValidConnection.i.on.t @ mailboxes.js:12
mailboxes.js:12 Reconnecting SlackRTM Socket for `679e0e62-177f-4a4b-86d0-1cb414f49ce1` Object
isValidConnection.i.on.t @ mailboxes.js:12
mailboxes.js:12 Reconnecting SlackRTM Socket for `f916bc85-a060-4d13-ba5a-946690862bea` Object
isValidConnection.i.on.t @ mailboxes.js:12
mailboxes.js:12 [SYNCS][WebSocket] Socket opened
```

**Comment by Thomas101** [comment]
Thanks for reporting. The only thing that I know can prolong sleep, is when a tab has child windows open, but you said this happens after restarting too. When it failed for you, did you just have the main Wavebox window open or did you have additional windows open too?

I'm struggling to reproduce it here, maybe I'm doing something different. Can you give this a try...

* Restart the app
* Right-click on the account that loaded in and then pick sleep (which account type is it? E.g. Slack, Gmail etc...)
* Does that bring up the "This account is sleeping" page or does something else happen?

**Comment by TheSpyder** [comment]
Sorry, what I mean is restarting wavebox does fix the issue but it eventually recurs. I just woke my laptop up after a few hours asleep and it's happened again (there is an extra window open, but it's from a service which was already asleep).

I use additional wavebox windows all the time - from what I've seen the main account service can sleep when it has child windows open?

**Comment by Thomas101** [comment]
Yep, you're right. It doesn't trigger when there are any secondary windows open. I think there will be another stable release over the next couple of days with this :)

**Comment by TheSpyder** [comment]
oh, so it doesn't trigger sleep for all accounts when _any_ secondary windows are open? ouch :)

I think that might be it. I ran a quick test and having a window open in the background does indeed seem to trigger the issue, closing it also allowed manual sleep to work. Along the same lines, I suggest either a permanent override specific to always allow manual sleeping or some kind of UI notification when a manual sleep does nothing.

I wonder whether I make heavier use of secondary windows than most. My usage pattern is fairly specific:

- links from work slack and gmail open in new account windows of jira, work google, notion (this happens a lot)
- I never use new windows for slack (multiple servers), home google, or discord

At some level I will find it useful for child windows to keep the parent open, but I _do_ regularly open those child windows from other accounts (which is why I love the new link rule feature).

**Comment by Thomas101** [comment]
I think I fixed a bug in the beta channel where the linking between windows isn't always carried across. This had the side-effect of making this bug more pronounced. It should be fixed in 4.7.1.

Also now when you manually trigger sleep it will ignore the child window check - depending on the relationship between the parent and child, it may cause windows to auto-close. (Gmail compose is one example of this).
